### PR TITLE
Removed spring dependency

### PR DESCRIPTION
**Pull request by ashwanidausodia**
Spring is no longer a dependency. However the the usage has changed a
little. Now interface is more like ForkJoinPool or other Executers in
Java Standard library. You create instance of AsyncRequestSerializer
passing the config. And then call submit method on that instance with
with (key, request) and get the Future back.

### CODE REVIEW

**Comment by ashwanidausodia** [comment]
It fixes both #2 and #3

File: src/main/java/com/samsung/lib/requestserializer/AsyncRequestSerializerConfig.java [hunk 1]
```diff
@@ -15,23 +15,60 @@
...
+  final int workerThreadPoolSize;
+  final int localRequestQueueTimeOut;
+
+  private AsyncRequestSerializerConfig(
+      int submitRetryCount,
+      int submitRetryDelay,
+      int workerThreadPoolSize,
+      int localRequestQueueTimeOut
+  ) {
+    this.submitRetryCount = submitRetryCount;
```
Comment (y-arun) [inline src/main/java/com/samsung/lib/requestserializer/AsyncRequestSerializerConfig.java]: Actually - we can minimize argument list by passing instance of Builder to this constructor. And initializing like
```java
private AsyncRequestSerializerConfig(Builder builder) {
    this.submitRetryDelay = builder.submitRetryDelay;
    ....
}
```

File: src/main/java/com/samsung/lib/requestserializer/AsyncRequestSerializerConfig.java [hunk 2]
```diff
@@ -15,23 +15,60 @@
...
-  public Integer workerThreadPoolSize;
+    public Builder setLocalRequestQueueTimeOut(int localRequestQueueTimeOut) {
+      this.localRequestQueueTimeOut = localRequestQueueTimeOut;
+      return this;
+    }
 
-  @Value(value = "#{config['asyncreqserializer.poolableworkerthread.timeout'] ?: '100'}")
-  public Integer localRequestQueueTimeOut;
+    public AsyncRequestSerializerConfig build() {
+      return new AsyncRequestSerializerConfig(
```
Comment (y-arun) [inline src/main/java/com/samsung/lib/requestserializer/AsyncRequestSerializerConfig.java]: If we follow above style this code can be further simplified as
```java
return new AsyncRequestSerializerConfig(this);
```

**Review by ghost — COMMENTED** [review]
There is one remark on using builder pattern based on "Effective Java, Chapter 2, pg 14" If you can update as per that recommendation. The changes is good to go. Thanks

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**